pkgs/os-specific/linux/kernel/linux-5.17.nix

```diff
@@ -0,0 +1,23 @@
+{ lib, buildPackages, fetchurl, perl, buildLinux, nixosTests, modDirVersionArg ? null, ... } @ args:
+
+with lib;
+
+buildLinux (args // rec {
+  version = "5.17";
+
+  # modDirVersion needs to be x.y.z, will automatically add .0 if needed
+  modDirVersion = if (modDirVersionArg == null) then concatStringsSep "." (take 3 (splitVersion "${version}.0")) else modDirVersionArg;
+
+  # branchVersion needs to be x.y
+  extraMeta.branch = versions.majorMinor version;
+
+  # src = fetchurl {
```

Comment:
Should this be here?

Also, looks like the `.tar.xz` version should work, based on glance at `kernel.org`'s suggested download URL.

Is there a reason to use the git snapshot instead?